src/libsystemd-network/icmp6-util.c

```diff
@@ -121,7 +121,7 @@ int icmp6_receive(
         /* This needs to be initialized with zero. See #20741. */
         CMSG_BUFFER_TYPE(CMSG_SPACE(sizeof(int)) + /* ttl */
                          CMSG_SPACE_TIMEVAL) control = {};
-        struct iovec iov = {};
+        struct iovec iov = { buffer, size };
```

Comment:
Why not to use IOVEC_NAKE()?

Reply:
Because it's a cast that reduces type safety. IOVEC_MAKE is useful when assigning to a variable so that it's not necessary to do the cast to `(struct iovec)` explicitly. But here it's not needed and I think the macro is not useful.

Reply:
OK. I prefer to use IOVEC_MAKE() even if it is not necessary. But, checking our code base, we actually do not use the macro sometimes. So, let's keep this as is at least now.